vpr/src/place/place.cpp

```diff
@@ -91,94 +95,106 @@ struct t_placer_statistics {
         sum_of_squares;
     int success_sum;
 };
+constexpr float INVALID_DELAY = std::numeric_limits<float>::quiet_NaN();
+constexpr double INVALID_COST = std::numeric_limits<double>::quiet_NaN();
 
-struct t_placer_costs {
-    //Although we do nost cost calculations with float's we
-    //use doubles for the accumulated costs to avoid round-off,
-    //particularly on large designs where the magnitude of a single
-    //move's delta cost is small compared to the overall cost.
-    double cost;
-    double bb_cost;
-    double timing_cost;
-};
+/*******************************************************************************
+ * Below is a list of definitions of data structures declared as `extern` in   *
+ * place_global.h. These variables were originally local to the current file.  *
+ * However, they were moved so as to facilitate moving some of the routines    *
+ * in the current file into other source files.                                *
+ *******************************************************************************/
```

Comment:
It seems like they were exported rather than moved. This seems to expose a lot of details; why not wrap these globals into a singleton (global) object? This will make it easier to duplicate these data structures e.g. to allow parallelism.

Reply:
I can wrap them into a single global structure and provide accessor functions for them (so that I don't have to declare them as extern), but I don't think there's a need to duplicate them, as they all serve different purposes and are only used by ```place.cpp`` once per VPR flow.

Reply:
After giving it more thought, I think I'll first leave it as it is in the current PR, and then refactor it into something similar to vpr_context.h in a new PR (something like g_place_ctx or g_place_structs). That will reduce the amount of clutter in the code.